app/src/main/java/com/kylecombes/micrepeater/MainActivity.java

```diff
@@ -99,11 +99,12 @@ public void registerBluetoothSCOListener() {
         // Register a callback to get notified about Bluetooth connect/disconnect events
         BluetoothStateReceiver.getInstance().registerStateChangeReceiver(
                 new BluetoothStateReceiver.StateChangeReceiver() {
-                    public void stateChanged(boolean deviceConnected, boolean scoAudioConnected, int batterylevel) {
+                    public void stateChanged(boolean deviceConnected, boolean scoAudioConnected,
+                            Integer batteryLevel) {
                         if (deviceConnected && !scoAudioConnected)
                             activateBluetoothSco();
                         mScoAudioConnected = scoAudioConnected;
```

Comment:
Do we need the variable `mScoAudioConnected` anymore?

Reply:
There's definitely an issue with the way we determine if a device is connected or not right now. For instance, we don't always get a "connected" notification/intent, so it's quite possible that we're able to use a connected Bluetooth device without it showing as "Connected". This is something that needs to be investigated and we need to determine what exactly we should be looking for.

Reply:
I took a pass at fixing this. There's still some weird behavior, especially if you have other Bluetooth devices that disconnect while the mic is in use. I wonder if there's a way to check whether the mic has disconnected. Maybe we should just be keeping track of whether Bluetooth SCO is available.